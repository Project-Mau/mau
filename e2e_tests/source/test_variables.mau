= Variables

:answer:42
:wikipedia_link:[link]("https://en.wikipedia.org/wiki/42_(number)")

This file tests variables. Above there is a definition of the variable `answer`, which is used in the next sentence.

The Answer to the Ultimate Question of Life, the Universe, and Everything is {answer}.

There is also a definition of a full link macro: {wikipedia_link}.

Interpolation can be blocked escaping the curly braces: \{answer}. This happens automatically in verbatim: `{answer}`.

:true:
:!false:

Mau supports booleans: {true} and {false}.

:value:5
:module.value:6

You can also use namespaces with a dotted syntax: {value} and {module.value}.
