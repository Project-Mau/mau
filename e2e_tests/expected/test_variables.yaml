data:
  args: []
  content:
  - data:
      anchor: variables-6933
      args: []
      kwargs: {}
      level: 1
      tags: []
      type: header
      value: Variables
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'This file tests variables. Above there is a definition of the
                variable '
          - data:
              type: verbatim
              value: answer
          - data:
              type: text
              value: ', which is used in the next sentence.'
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: The Answer to the Ultimate Question of Life, the Universe, and
                Everything is 42.
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'There is also a definition of a full link macro: '
          - data:
              target: https://en.wikipedia.org/wiki/42_(number)
              text: https://en.wikipedia.org/wiki/42_(number)
              type: link
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'Interpolation can be blocked escaping the curly braces: {answer}.
                This happens automatically in verbatim: '
          - data:
              type: verbatim
              value: '{answer}'
          - data:
              type: text
              value: .
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'Mau supports booleans: True and False.'
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  - data:
      args: []
      content:
        data:
          content:
          - data:
              type: text
              value: 'You can also use namespaces with a dotted syntax: 5 and 6.'
          type: sentence
      kwargs: {}
      tags: []
      type: paragraph
  kwargs: {}
  tags: []
  type: document

